tools/ddrgen/src/scanners/dwarf/AixDwarfParser.cpp

```diff
@@ -0,0 +1,1034 @@
+		ret = DW_DLV_NO_ENTRY;
+	} else {
+		/* Get the next sibling of a Die. */
+		*dieOut = die->_sibling;
+	}
+	return ret;
+}
+
+int
+dwarf_finish(Dwarf_Debug dbg, Dwarf_Error *error)
+{
+	/* Free all Dies and Attributes created in dwarf_init(). */
+	_currentCU = _firstCU;
+	while (NULL != _currentCU) {
+		Dwarf_CU_Context *nextCU = _currentCU->_nextCU;
+		if (NULL != _currentCU->_die) {
+			deleteDie(_currentCU->_die);
+		}
+		delete(_currentCU);
+		_currentCU = nextCU;
+	}
+	_fileList.clear();
+	_currentCU = NULL;
+	_firstCU = NULL;
+	return DW_DLV_OK;
+}
+
+int
+dwarf_init()
+{
+	int ret = DW_DLV_OK;
+	FILE *fp = NULL;
+	char *buffer = NULL;
+	size_t len = 0;
+	
+	/* Initialize some values */	
+	_fileID = 0;
+	_startNewFile = 0;
+
+	/* Open up the input file. */
+	fp = fopen ("aixoutput", "r");
+	
+	/* Parse through the file */
+	while (-1 != getline(&buffer, &len, fp)) {
+		ret = parseDwarfInfo (buffer);
+	}
+	
+	return ret;
+}
+
+static void
+deleteDie(Dwarf_Die die)
+{
+	/* Free the Die's attributes. */
+	Dwarf_Attribute attr = die->_attribute;
+	while (NULL != attr) {
+		Dwarf_Attribute next = attr->_nextAttr;
+		if (NULL != attr->_stringdata) {
+			free(attr->_stringdata);
+		}
+		delete(attr);
+		attr = next;
+	}
+
+	/* Free the Die's siblings and children. */
+	if (NULL != die->_sibling) {
+		deleteDie(die->_sibling);
+	}
+	if (NULL != die->_child) {
+		deleteDie(die->_child);
+	}
+	delete(die);
+}
+
+
+/**
+ * Parses the string to file ID. Extracts the integer betwen the square brackets.
+ *
+ * @param[in] data: string from which to extract ID from. The expected format is "[####]"
+ *
+ * @return file ID as double
+ */
+double
+extractFileID(const string data)
+{
+	stringstream ss;
+	double ret = 0;
+	ss << data.substr(1, data.length() - 1);
+	ss >> ret;
+	return ret;
+}
+
+/**
+ * Extracts the typeID from the name string that's passed in
+ *
+ * @param data: string that needs to be parsed for typeID EX: (T|t) INTEGER
+ *
+ * @return typeID as double
+ */
+double
+extractTypeID(const string data, const int index)
+{
+	stringstream ss;
+	double ret = 0;
+	ss << data.substr(index);
+	ss >> ret;
+	return ret;
+}
+
+/**
+ * Checks the first index of the field declaration string for compiler generated tag, GenSpec.
+ *	Possible valid formats, that could potentially return true are:
+ *		GenSpec AccessSpec AnonSpec DataMember
+ *		GenSpec VirtualSpec AccessSpec OptVirtualFuncIndex MemberFunction
+ *
+ * @param[in] data: EX: cup5__vfp:__vfp:23,0,32;uN222;uN223;vu1[di:__dt__Q3_3std7_LFS_ON10ctype_baseFv:2;; << This would return true
+ *
+ * @return boolean value indicating if the declaration is compiler generated or not
+ */
+bool
+isCompilerGenerated(const string data)
+{
+	return data.empty() ? false : ('c' == data[0]);
+}
+
+ /**
+ * Returns type of bound
+ *
+ * @param[in] data: bound declaration string
+ *
+ * @return bound type based on the data string
+ */
+string
+parseBound(const string data)
+{
+	string ret = "INVALID";
```

Comment:
Just a generate note on error handling: 
- If a function can fail, it should have a return type of DDR_RC and return either DDR_RC_OK or DDR_RC_ERROR. Other returned data can then be by parameters.
- Function return types for the functions you create should always be checked. If DDR_RC_ERROR was returned, execution should halt and not continue to make additional function calls. Note that our coding standard is still to have only one "return" per function.
- If a function calls any functions which returns DDR_RC, then it itself can fail and it should also return a DDR_RC.

I would suggest checking some of the DwarfScanner.cpp functions for error handling style if you are unsure.